build-support/bin/native/calculate_engine_hash.sh

```diff
@@ -10,11 +10,14 @@ source "${REPO_ROOT}/build-support/common.sh"
 
 readonly NATIVE_ROOT="${REPO_ROOT}/src/rust/engine"
 
-# N.B. Set $MODE to "debug" for faster builds.
-readonly MODE="${MODE:-release}"
+# N.B. Set $MODE to "release" for much faster Pants runs, but slower compiles.
+# We optimize for faster compiles to reduce the barrier to entry for new contributors
+# (https://github.com/pantsbuild/pants/issues/9822). Most core Pants devs will want to permanently
```

Comment:
> Most core Pants devs will want to permanently

It's more subtle than that I think. If you're primarily editing the Python code, then yes, maybe. If you're editing the rust code, then you'll spend more time in MODE=debug.

Reply:
Oh: one other thing that might be helpful here would be to render something on stderr when pants is being run in debug mode?

Reply:
> to render something on stderr when pants is being run in debug mode?

Maybe - I'm not sure how I feel about that. On one hand, it addresses contributors possibly asking "Why is Pants so slow?!" 

On the other, it's noisy and presumably couldn't be turned off. For most contributors doing 1 or 2 smaller patches—like Josh simply trying to run `./v2 fmt` this morning—I could see it being annoying and more cognitive overload than necessary.